3.10/indexing-working-with-indexes.md

```diff
@@ -82,34 +82,59 @@ Note that `_key` implicitly has an index assigned to it.
 
 <!-- arangod/V8Server/v8-vocindex.cpp -->
 
-ensures that an index exists
+Ensures that an index exists:
+
 `collection.ensureIndex(index-description)`
 
 Ensures that an index according to the *index-description* exists. A
 new index will be created if none exists with the given description.
 
-The *index-description* must contain at least a *type* attribute.
+Calling this method returns an index object. Whether or not the index
+object existed before the call is indicated in the return attribute
+*isNewlyCreated*.
+
+The *index-description* input value must contain at least a *type* attribute.
 Other attributes may be necessary, depending on the index type.
 
 **type** can be one of the following values:
 - *persistent*: persistent index
 - *fulltext*: fulltext index (deprecated from ArangoDB 3.10 onwards)
 - *geo*: geo index, with _one_ or _two_ attributes
 
+**fields** is an array of attribute paths, containing the document attributes
+(or subattributes) to be indexed. Some indexes allow using only a single path,
+and others allow multiple. 
+If multiple attributes are used, their order matters.
+
+If an attribute path contains an `[*]` extension (e.g. `friends[*].id`), it means
+that the index attribute value is treated as an array and all array members are
+indexed separately. This is possible with *persistent* indexes.
+
+**storedValues**: in indexes of type *persistent*, additional attributes can be
+stored in the index. These additional attributes cannot be used for
+index lookups or for sorting, but they can be used for projections. This allows an
+index to fully cover more queries and avoid extra document lookups.
+The maximum number of attributes in **storedValues** is 32.
+It is not possible to create multiple indexes with the same **fields** attributes
+and uniqueness but different **storedValues** attributes. That means the value of 
+**storedValues** is not considered by index creation calls when checking if an 
+index is already present or needs to be created.
```

Comment:
Suggest to add how we treat missing attributes. And mention that `storedValues` are not checked for uniqueness if `unique` is set. Just to be more explicit for end-user.

Reply:
I pushed a change that addresses this.